samples/attestations/verify_attestation_chains.py

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env python
+
+# Copyright 2021 Google, LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This application verifies HSM certificate chains.
+
+For more information, visit https://cloud.google.com/kms/docs/attest-key.
+"""
+
+# [START kms_verify_chains]
+import argparse
+import gzip
+import io
+import zipfile
+
+from cryptography import exceptions
+from cryptography import x509
+from cryptography.hazmat import backends
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.asymmetric import padding
+import pem
+import requests
+
+ATTESTATION_SIGNATURE_LEN = 256
+
+MANUFACTURER_CERT_URL = 'https://www.marvell.com/content/dam/marvell/en/public-collateral/security-solutions/liquid_security_certificate.zip'
+
+# <Name(C=US,ST=California,L=San Jose,O=Cavium\, Inc.,OU=LiquidSecurity,CN=localca.liquidsecurity.cavium.com)>
+MANUFACTURER_CERT_SUBJECT_BYTES = (
+    b'0\x81\x911\x0b0\t\x06\x03U\x04\x06\x13\x02US1\x130\x11\x06\x03U\x04\x08'
+    b'\x0c\nCalifornia1\x110\x0f\x06\x03U\x04\x07\x0c\x08San Jose1\x150\x13\x06'
+    b'\x03U\x04\n\x0c\x0cCavium, Inc.1\x170\x15\x06\x03U\x04\x0b\x0c\x0e'
+    b'LiquidSecurity1*0(\x06\x03U\x04\x03\x0c!localca.liquidsecurity.cavium.com'
+)
+
+# The owner root cert can be obtained from
+# 'https://www.gstatic.com/cloudhsm/roots/global_1498867200.pem'
+OWNER_ROOT_CERT_PEM = """Certificate:
```

Comment:
We dynamically fetch the manufacturer certificate, but we embed the owner root certificate. Is there a reason why we treat them differently?

Reply:
We could also embed the manufacturer root certificate, but that would put the burden of verifying that the certificate we embed is the same as the actual manufacturer root certificate on the user. On the other hand, we could also fetch the Google/owner root certificate dynamically, but I don't think this would provide much benefit because we are the ones publishing this script.

Reply:
Ok